trunk/client.h

```diff
@@ -419,11 +419,15 @@ typedef enum {
 typedef struct {
 	qboolean initialized;
 
+    // If true files will be extracted into a temporary directory.  If false,
+    // files will be extracted into the same directory as the dzip file.
+    qboolean use_temp_dir;
+
 	// Directory into which dzip files will be extracted.
-	char extract_dir[MAX_OSPATH];
```

Comment:
`MAX_OSPATH` is pretty short (128).  Even worse, there's a potential overflow in this line:
```c
COM_StripExtension (tempdir, ctx->dem_path);
```
Since `sizeof(tempdir)` is 1024, this command could overrun `dem_path`'s buffer.  Using the same buffer size throughout should avoid this.